src/parse/text/simple_extensions/argument.rs

```diff
@@ -0,0 +1,633 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use thiserror::Error;
+
+use crate::{
+    parse::{Context, Parse},
+    text::simple_extensions,
+};
+
+/// A parsed [simple_extensions::ArgumentsItem].
+#[derive(Clone, Debug)]
+pub enum ArgumentsItem {
+    /// Arguments that support a fixed set of declared values as constant arguments.
+    EnumArgument(EnumerationArg),
+
+    /// Arguments that refer to a data value.
+    ValueArgument(ValueArg),
+
+    /// Arguments that are used only to inform the evaluation and/or type derivation of the function.
+    TypeArgument(TypeArg),
+}
+
+impl ArgumentsItem {
+    #[inline]
+    fn parse_name(name: Option<String>) -> Result<Option<String>, ArgumentsItemError> {
+        match name {
+            Some(s) if s.is_empty() => {
+                Err(ArgumentsItemError::EmptyOptionalField("name".to_string()))
+            }
+            _ => Ok(name),
+        }
+    }
+
+    #[inline]
+    fn parse_description(
+        description: Option<String>,
+    ) -> Result<Option<String>, ArgumentsItemError> {
+        match description {
+            Some(s) if s.is_empty() => Err(ArgumentsItemError::EmptyOptionalField(
+                "description".to_string(),
+            )),
+            _ => Ok(description),
+        }
+    }
```

Comment:
Can we merge this and pass the name for the error as argument?

Reply:
Extracted the shared logic into a new function: [commit](https://github.com/substrait-io/substrait-rs/pull/185/commits/80873cf325f1bb026355f627003421399c9b48dc).